docs: adds documentation for assign code author reviewer (#194)
reviewpad/docs: docs/guides/built-ins/actions/assignCodeAuthorReviewer/index.mdx

import CodeBlock from "@theme/CodeBlock";
import Example from "!!raw-loader!./example.yml";

**Description**:

> Intelligently assigns a reviewer based on the number of lines a user has contributed to in the pull requests changed files.
> This action will always pick a user different than the author of a pull request.
> 
> If a reviewer could not be determined intelligently, this action will assign a random user.
> 
> However, if the pull request already has a reviewer, nothing happens. This is to prevent adding a reviewer each time the pull request is updated.

Reviewer: Maybe we could take advantage of the definition provided in https://github.com/reviewpad/reviewpad/issues/678.
WDYT @ferreiratiago ?
```suggestion
Intelligently assigns the most relevant and available reviewer to the pull request.

The relevancy is based on the code ownership information available on the impacted source files. The more lines of code are owned by a reviewer on the impacted source files, the more relevant this reviewer is for the pull request.

The availability is based on the number of open pull requests per reviewer. Above a configurable threshold (`max_reviews`), the reviewer is considered unavailable.

The assigned reviewer will always be different than the pull request's author.

If no relevant reviewer is available for this pull request, a random user of the GitHub organization will be assigned as the reviewer.

If the pull request already has a reviewer, no action will be taken.

When there is no reviewer to assign to, an error is returned.
```

:::caution

This built-in is not available for [`issue`](/guides/syntax#workflow).

:::

**Parameters**:

> | Parameter                       | Type     | Description                                                               |
> | :------------------------------ | :--------| :------------------------------------------------------------------------ |
> | `max_reviews` (optional)        | int      | The maximum number of open reviews a user can have, no limit by default.  |
> | `excluded_reviewers` (optional) | []string | A list of usernames to exclude from review requests.                      |

Reviewer: Minor suggestion:
```suggestion
| Parameter                       | Type     | Description                                                                       |
| :------------------------------ | :--------| :-------------------------------------------------------------------------------- |
| `max_reviews` (optional)        | int      | The maximum number of open reviews a user can have. By default, no limit is set.  |
| `excluded_reviewers` (optional) | []string | List of usernames to exclude from review requests.                                |
```

The relevancy is based on the code ownership information available on the impacted source files. The more lines of code are owned by a reviewer on the impacted source files, the more relevant this reviewer is for the pull request.

The availability is based on the number of open pull requests per reviewer. Above a configurable threshold (`max_reviews`), the reviewer is considered unavailable.

The assigned reviewer will always be different than the pull request's author.

If no relevant reviewer is available for this pull request, a random user of the GitHub organization will be assigned as the reviewer.

If the pull request already has a reviewer, no action will be taken.

When there is no reviewer to assign to, an error is returned.

:::caution

This built-in is not available for [`issue`](/guides/syntax#workflow).

:::

**Parameters**:

| Parameter                             | Type     | Description                                                                      |
| :------------------------------------ | :--------| :------------------------------------------------------------------------------- |

> | `total_required_reviewers` (optional) | int      | The total number of required reviewers.                                          |

Reviewer: ```suggestion
| `total_required_reviewers` (optional) | int       | The total number of required reviewers. By default, it is one.                    |
```

| `excluded_reviewers` (optional)       | []string | List of usernames to exclude from review requests.                               |

> | `max_reviews` (optional)              | int      | The maximum number of open reviews a user can have. By default, no limit is set. |

Reviewer: What do you think?
```suggestion
| `max_reviews` (optional)              | int       | The maximum number of open pull requests that a single user can be responsible for reviewing.  |
```

> | `max_reviews` (optional)              | int      | The maximum number of open pull requests that a single user can be responsible for reviewing |

Reviewer: Just missing a dot 🙂 
```suggestion
| `max_reviews` (optional)              | int      | The maximum number of open pull requests that a single user can be responsible for reviewing. |
```
